autogpt_platform/backend/backend/server/rest_api.py

```diff
@@ -1,695 +1,117 @@
-
-        latest_version_graph = next(
-            v for v in existing_versions if v.version == latest_version_number
-        )
-        if latest_version_graph.is_template != graph.is_template:
-            raise HTTPException(
-                400, detail="Changing is_template on an existing graph is forbidden"
-            )
-        graph.is_active = not graph.is_template
-        graph.reassign_ids()
-
-        new_graph_version = await graph_db.create_graph(graph, user_id=user_id)
-
-        if new_graph_version.is_active:
-            # Ensure new version is the only active version
-            await graph_db.set_graph_active_version(
-                graph_id=graph_id, version=new_graph_version.version, user_id=user_id
-            )
-
-        return new_graph_version
+        is_template=False,
+    ):
+        return await backend.server.routers.v1.create_new_graph(create_graph, user_id)
 
-    @classmethod
-    async def set_graph_active_version(
-        cls,
-        graph_id: str,
-        request_body: SetGraphActiveVersion,
-        user_id: Annotated[str, Depends(get_user_id)],
+    @staticmethod
+    async def test_get_graph_run_status(
+        graph_id: str, graph_exec_id: str, user_id: str
     ):
-        new_active_version = request_body.active_graph_version
-        if not await graph_db.get_graph(graph_id, new_active_version, user_id=user_id):
-            raise HTTPException(
-                404, f"Graph #{graph_id} v{new_active_version} not found"
-            )
-        await graph_db.set_graph_active_version(
-            graph_id=graph_id,
-            version=request_body.active_graph_version,
-            user_id=user_id,
+        return await backend.server.routers.v1.get_graph_run_status(
+            graph_id, graph_exec_id, user_id
         )
 
-    def execute_graph(
-        self,
-        graph_id: str,
-        node_input: dict[Any, Any],
-        user_id: Annotated[str, Depends(get_user_id)],
-    ) -> dict[str, Any]:  # FIXME: add proper return type
-        try:
-            graph_exec = self.execution_manager_client.add_execution(
-                graph_id, node_input, user_id=user_id
-            )
-            return {"id": graph_exec["graph_exec_id"]}
-        except Exception as e:
-            msg = e.__str__().encode().decode("unicode_escape")
-            raise HTTPException(status_code=400, detail=msg)
-
-    async def stop_graph_run(
-        self, graph_exec_id: str, user_id: Annotated[str, Depends(get_user_id)]
-    ) -> list[execution_db.ExecutionResult]:
-        if not await execution_db.get_graph_execution(graph_exec_id, user_id):
-            raise HTTPException(
-                404, detail=f"Agent execution #{graph_exec_id} not found"
-            )
-
-        await asyncio.to_thread(
-            lambda: self.execution_manager_client.cancel_execution(graph_exec_id)
+    @staticmethod
+    async def test_get_graph_run_node_execution_results(
```

Comment:
why is it renamed?

Reply:
To mark it clearly as a function that is only used during testing. It is not used when running the system